pkgs/README.md

```diff
@@ -510,6 +510,11 @@ In the interest of keeping our maintenance burden and the size of Nixpkgs to a m
 
 If a patch is available online but does not cleanly apply, it can be modified in some fixed ways by using additional optional arguments for `fetchpatch`. Check [the `fetchpatch` reference](https://nixos.org/manual/nixpkgs/unstable/#fetchpatch) for details.
 
+> [!Note]
+> The version of the package should not be changed just because a patch is applied. Nix can distinguish package changes based on the hash of the derivation generated by the expression. Users will still receive the patched package regardless of the current version they have installed.
+>
+> See [Versioning](#versioning) for details on package versioning.
+
```

Comment:
```suggestion
> [!Note]
> The version of the package does not need to be changed just because a patch is applied. Declarative package installations don't depend on the version, while imperative `nix-env` installations can use [`upgrade --eq/leq/--always`](https://nix.dev/manual/nix/2.25/command-ref/nix-env/upgrade#flags).
>
> See [Versioning](#versioning) for more information on package versioning.
```

This seems more like it. Should not -> does not need to, because one might still want to increase the version number to reach more people, especially because `nix-env upgrade` by default doesn't upgrade if the version stayed the same.
 
And I'd move this note one section up, since it's not directly related to _fetching_ of patches.

Reply:
Great thank you. I wasn't aware of the distinction in `nix-env`. What are the rules if you do wish to increase the version? From what I've read elsewhere, our package version should match upstream. Perhaps this is something to be added to the `#versioning` section?

Reply:
I guess that could be done in another PR, seems fairly orthogonal :)